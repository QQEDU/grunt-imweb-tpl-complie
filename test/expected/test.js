describe('static', function () {
    it('should able to render static html', function () {
        window['JST_tpl']['static']().should.equal('<p>Hello, world</p>');
    });

    it('should able to render with data', function () {
        window['JST_tpl']['render']({
            say: 'Hello, world'
        }).should.equal('<p>Hello, world</p>');
    });

    it('should able to render use it', function () {
        window['JST_tpl']['it']({
            say: 'Hello, world'
        }).should.equal('<p>Hello, world</p>');
    });

    it('should able to encode', function () {
        window['JST_tpl']['encode']({
            say: '<p>Hello, world</p>'
        }, {
            encodeHtml: function (str) {
                return (str + '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/`/g, '&#96;').replace(/'/g, '&#39;').replace(/"/g, '&quot;');
            }
        }).should.equal('&lt;p&gt;Hello, world&lt;/p&gt;');
    });

    it('should able to include template', function () {
        window['JST_tpl']['include']({ 
            items: [{ say: 'hello' }, { say: 'tencent' }, { say: 'qqedu' }] 
        }, {
            encodeHtml: function (str) {
                return (str + '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/`/g, '&#96;').replace(/'/g, '&#39;').replace(/"/g, '&quot;');
            }
        }).should.equal('<p>hello</p><p>tencent</p><p>qqedu</p>');
    });
});