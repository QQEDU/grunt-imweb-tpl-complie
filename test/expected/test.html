<html>
	<head>
	<meta charset="utf-8">
	<link rel="stylesheet" href="mocha.css" />
	</head>
	<body>
	<div id="mocha"></div>
	<script src="mocha.js"></script>
	<script src="should.js"></script>
	<script>
		mocha.ui('bdd');
		mocha.reporter('html');
	</script>
	<script src="js/1.js"></script>
	<script src="static_test.js"></script>
	<script>
	if (window.mochaPhantomJS) { mochaPhantomJS.run(); }
	else { mocha.run(); }
	</script>
	</body>
</html>